### PR TITLE
Fix ping permission issue and ssh is not active when retry many times to check nic status.

### PR DESCRIPTION
**Pull request by LiliDeng**

### CODE REVIEW

File: lisa/tools/ping.py [hunk 1]
```diff
@@ -29,7 +37,7 @@ def ping_async(
         if package_size:
             args += f" -s {package_size}"
 
-        return self.run_async(args, force_run=True)
+        return self.run_async(args, force_run=True, sudo=sudo)
```
Comment (squirrelsc) [inline lisa/tools/ping.py:40]: keep arguments consistent between sync and async v3ersion. if the sudo must be True, remove it from async method, and set it always True.

File: lisa/tools/ping.py [hunk 2]
```diff
@@ -48,7 +57,17 @@ def ping(
             count=count,
             interval=interval,
             package_size=package_size,
+            sudo=sudo,
         ).wait_result()
+        if not sudo and self._no_permission_pattern.findall(result.stdout):
```
Comment (squirrelsc) [inline lisa/tools/ping.py:65]: Sorry, I haven't noticed, this logic called ping twice. Is there any reason to call it second time? Can you add comments in code?

Comment (LiliDeng) [reply lisa/tools/ping.py:65]: For some distro, it needs sudo, on these distros, will retry with sudo.

Comment (squirrelsc) [reply lisa/tools/ping.py:65]: Please add it in code comments.

**Review by squirrelsc — APPROVED** [review]

**Pull request merged**